### DISCUSSION TITLE
Extracting numerals from text

### DISCUSSION THREAD

**Discussion by attilanagy234** [huspacy/huspacy#65] [Ideas]
**Is your feature request related to a problem? Please describe.**
I'm trying to extract numerals of different form from Hungarian text (percentages, dates, etc.). I was wondering if this is supported in Huspacy.

**Describe the solution you'd like**
Here's a small example I've been experimenting with:

```python
text = "A Magyar Fejlesztési Bank Rt. értékesítette 94,521 százalékos részvénycsomagját a BÁV Bizományi Kereskedőház és Záloghitel Rt-ben. A BÁV fő - 48,26 százalékos - tulajdonosa a kormány gyorsforgalmi úthálózat-fejlesztési programjában meghatározó szerepet játszó Vegyépszer Rt.-hez köthető Pro-Cash Rt. lett, az OTP Bank Rt.-nek a papírok 46,25 százaléka jutott - közölte a Népszabadság."

doc = nlp(text)
for ent in doc.ents:
    print(f"{ent}, {ent.label_}")
```

which results in:
```bash
Magyar Fejlesztési Bank Rt., ORG
BÁV Bizományi Kereskedőház és Záloghitel Rt-ben, ORG
BÁV, ORG
Vegyépszer Rt.-hez, ORG
Pro-Cash Rt., ORG
OTP Bank Rt.-nek, ORG
Népszabadság, ORG
```

If I run the same with the English model, I can extract percentages, dates, etc. as part of the entities as well.

**Describe alternatives you've considered**
I wrote some regexes already to extract the numerals, but this way my recall is expected to be lower (as dates & times can take many forms).

I can work with this, but it would be great to have this as part of the regular Huspacy NLP pipeline. 

I have not checked if for English this feature is part of the NER tagger or something else. We may be limited by the lack of training data.

**Additional context**
Versions:
- huspacy==0.9.0
- spacy==3.5.3
- model: hu_core_news_lg

**Comment by oroszgy** [comment]
@attilanagy234 thanks for your report. Unfortunately, the current system is limited by the training data available. One option could be to rely on spaCy's Matcher and the LIKE_NUM attribute. Another option to consider is to use [this](https://huggingface.co/novakat/nerkor-cars-onpp-hubert) alternative NER model for which we provide a simple integration: https://huspacy.github.io/recipes/nerpp/

**Reply by attilanagy234** [reply]
I tried using the `LIKE_NUM ` attribute, but it wasn't very accurate. Thanks for the tip on the `nerpp` model, I didn't know about this. I did a quick test and it seems to work pretty decently on my use-case. E.g. for the above example I get:
```
Magyar Fejlesztési Bank Rt., ORG
94,521 százalékos, PERCENT
BÁV Bizományi Kereskedőház és Záloghitel Rt-ben, ORG
BÁV, ORG
48,26 százalékos, PERCENT
Vegyépszer Rt.-hez, ORG
Pro-Cash Rt., ORG
OTP Bank Rt.-nek, ORG
46,25 százaléka, PERCENT
Népszabadság, MEDIA
```
I think I can work with this, thanks a lot @oroszgy 🙏🏼
